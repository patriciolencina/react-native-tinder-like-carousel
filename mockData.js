export const userJSONString =
  '{"results":[{"user":{"gender":"male","name":{"title":"mr","first":"reginald","last":"burton"},"location":{"street":"5668 andreas ave","city":"duncanville","state":"iowa","zip":"69310"},"email":"reginald.burton77@example.com","username":"beautifultiger643","password":"calgary","salt":"o50omLUi","md5":"59f4e2db81840e41eaa1b2d39f2ead55","sha1":"f1d65fd8f818137c5bc3f4bc147bb8ea109d5867","sha256":"b5a525ca0e449c81029829d7b7eca752fce49b696c58363b9e36cfd272f18e5f","registered":"1246053519","dob":"144374909","phone":"(712)-125-8909","cell":"(899)-446-4844","SSN":"707-51-3358","picture":"http://api.randomuser.me/portraits/men/61.jpg"},"seed":"11059646270402eb","version":"0.4"}]}';

export const userData = {
  gender: 'male',
  name: {
    title: 'mr',
    first: 'reginald',
    last: 'burton',
  },
  location: {
    street: '5668 andreas ave',
    city: 'duncanville',
    state: 'iowa',
    zip: '69310',
  },
  email: 'reginald.burton77@example.com',
  username: 'beautifultiger643',
  password: 'calgary',
  salt: 'o50omLUi',
  md5: '59f4e2db81840e41eaa1b2d39f2ead55',
  sha1: 'f1d65fd8f818137c5bc3f4bc147bb8ea109d5867',
  sha256: 'b5a525ca0e449c81029829d7b7eca752fce49b696c58363b9e36cfd272f18e5f',
  registered: '1246053519',
  dob: '144374909',
  phone: '(712)-125-8909',
  cell: '(899)-446-4844',
  SSN: '707-51-3358',
  picture: 'http://api.randomuser.me/portraits/men/61.jpg',
};
