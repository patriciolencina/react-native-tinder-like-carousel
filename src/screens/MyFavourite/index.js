import React from 'react';
import {reverse} from 'ramda';
import {useSelector} from 'react-redux';
import {getFavoriteDataPolished} from '../../hooks/user-data-process';
import MyFavouriteView from './view';

const MyFavourite = () => {
  const users = reverse(useSelector(getFavoriteDataPolished));
  return <MyFavouriteView users={users} />;
};
export default MyFavourite;
