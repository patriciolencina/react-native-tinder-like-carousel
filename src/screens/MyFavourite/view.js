import React from 'react';
import {
  View,
  Text,
  FlatList,
  Dimensions,
  StyleSheet,
  Image,
  SafeAreaView,
} from 'react-native';

const MyFavoriteView = ({users}) => {
  return (
    <SafeAreaView style={{flex: 1, backgroundColor: 'white'}}>
      <FlatList
        data={users}
        numColumns={1}
        keyExtractor={item => item.username}
        renderItem={({item}) => (
          <View
            style={{
              backgroundColor: 'white',
              flexDirection: 'row',
              borderBottomColor: 'gray',
              borderBottomWidth: 1,
              padding: 20,
            }}>
            <Image
              resizeMode="contain"
              source={{uri: item.picture}}
              style={{
                height: 50,
                width: 50,
                borderRadius: 25,
                borderColor: 'gray',
                borderWidth: 1,
                marginRight: 20,
              }}
            />
            <View
              style={{
                flexDirection: 'column',
              }}>
              <Text style={{color: 'black'}}>{item.username}</Text>
              <Text style={{color: 'black'}}>{item.address}</Text>
            </View>
          </View>
        )}
      />
    </SafeAreaView>
  );
};

export default MyFavoriteView;
