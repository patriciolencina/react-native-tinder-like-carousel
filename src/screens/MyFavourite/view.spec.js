import React from 'react';

import {shallow, ShallowWrapper} from 'enzyme';
import {SafeAreaView, Flatlist} from 'react-native';
import MyFavourite from './view';
import {userData} from '../../../mockData';

const createTestProps = (props: Object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  users: [userData],
  ...props,
});

describe('MyFavourite', () => {
  describe('rendering', () => {
    let wrapper: ShallowWrapper;
    let props: any; // use type "any" to opt-out of type-checking
    beforeEach(() => {
      props = createTestProps({});
      wrapper = shallow(<MyFavourite {...props} />); // no compile-time error
    });

    it('should render with Flatlist with user data', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should render a <SafeAreaView /> with no user data', () => {
      expect(shallow(<MyFavourite />).find(SafeAreaView)).toHaveLength(1);
    });
  });
});
