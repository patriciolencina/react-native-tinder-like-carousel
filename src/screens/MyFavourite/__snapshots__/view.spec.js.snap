// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFavourite rendering should render with Flatlist with user data 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <FlatList
    data={
      Array [
        Object {
          "SSN": "707-51-3358",
          "cell": "(899)-446-4844",
          "dob": "144374909",
          "email": "reginald.burton77@example.com",
          "gender": "male",
          "location": Object {
            "city": "duncanville",
            "state": "iowa",
            "street": "5668 andreas ave",
            "zip": "69310",
          },
          "md5": "59f4e2db81840e41eaa1b2d39f2ead55",
          "name": Object {
            "first": "reginald",
            "last": "burton",
            "title": "mr",
          },
          "password": "calgary",
          "phone": "(712)-125-8909",
          "picture": "http://api.randomuser.me/portraits/men/61.jpg",
          "registered": "1246053519",
          "salt": "o50omLUi",
          "sha1": "f1d65fd8f818137c5bc3f4bc147bb8ea109d5867",
          "sha256": "b5a525ca0e449c81029829d7b7eca752fce49b696c58363b9e36cfd272f18e5f",
          "username": "beautifultiger643",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</SafeAreaView>
`;
